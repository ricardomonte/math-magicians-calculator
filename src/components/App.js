import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

function App() {
  const [state, setState] = useState({ total: null, next: null, operator: null });

  const handleClick = (simbolOperation) => {
    setState(Calculate(state, simbolOperation));
  };

  const result = () => {
    const { total, next, operator } = state;
    let display = '0';
    if (next) {
      display = total + operator + next;
    } else if ((operator && operator !== '=') && total) {
      display = total + operator;
    } else {
      display = total || '0';
    }
    return display;
  };

  return (
    <>
      <div className="Math">
        <h2>Let&acute;s do some math </h2>
      </div>
      <div className="w-100 h-100 App">
        <Display calculation={result()} />
        <ButtonPanel onClick={handleClick} />
      </div>
    </>
  );
}

export default App;
