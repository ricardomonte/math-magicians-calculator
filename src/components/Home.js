import React from 'react';
import { LoremIpsum } from 'react-lorem-ipsum';

function Home() {
  return (
    <div className="Home">
      <h1>Welcome to our page</h1>
      <LoremIpsum p={2} />
    </div>
  );
}

export default Home;
