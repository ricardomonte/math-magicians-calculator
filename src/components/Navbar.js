import React from 'react';
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <div className="Navbar">
      <h1>Math Magicians</h1>
      <nav>
        <Link to="/">Home</Link>
        <Link to="/calculator">Calculator</Link>
        <Link to="/quotes">Quotes</Link>
      </nav>
    </div>
  );
}

export default Navbar;
