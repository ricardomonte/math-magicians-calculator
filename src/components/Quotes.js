import React from 'react';

function Quotes() {
  return (
    <div className="Quote">
      <figure>
        <blockquote>
          &ldquo;Pure mathematics is, in its way, the poetry of logic ideas&rdquo;
        </blockquote>
        <figcaption>-Albert Einstein-</figcaption>

      </figure>
    </div>
  );
}

export default Quotes;
