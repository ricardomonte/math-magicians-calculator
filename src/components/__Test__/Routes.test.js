import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../Navbar';
import Home from '../Home';
import App from '../App';
import Quotes from '../Quotes';

test('return content from Routes matches snapshot', () => {
  const component = render(
    <BrowserRouter>
      <>
        <Navbar />
        <Switch>
          <Route path="/" component={Home} exact />
          <Route path="/calculator" component={App} />
          <Route path="/quotes" component={Quotes} />
        </Switch>
      </>
    </BrowserRouter>,
  );
  expect(component.container).toMatchSnapshot();
});

test('content will render without crash', () => {
  const div = document.createElement('div');
  render(
    <BrowserRouter>
      <>
        <Navbar />
        <Switch>
          <Route path="/" component={Home} exact />
          <Route path="/calculator" component={App} />
          <Route path="/quotes" component={Quotes} />
        </Switch>
      </>
    </BrowserRouter>, div,
  );
});
