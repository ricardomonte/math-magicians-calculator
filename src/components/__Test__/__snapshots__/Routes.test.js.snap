// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return content from Routes matches snapshot 1`] = `
<div>
  <div
    class="Navbar"
  >
    <a
      class="logo"
      href="/"
    >
      Math Magicians
    </a>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/calculator"
      >
        Calculator
      </a>
      <a
        href="/quotes"
      >
        Quotes
      </a>
    </nav>
  </div>
  <div
    class="Home"
  >
    <h1>
      Welcome to our page
    </h1>
    <p>
      Lorem ipsum odor amet, consectetuer adipiscing elit. Facilisi sollicitudin purus auctor at cursus amet auctor lobortis velit. Fames nisi ullamcorper gravida cubilia conubia accumsan vitae sollicitudin. Quisque vulputate habitasse libero dapibus primis facilisis elementum. Quam est ex porttitor senectus congue magna lacinia. Morbi torquent convallis rutrum sit, dui cubilia quisque venenatis. Velit nullam aliquam posuere himenaeos felis proin magnis ultricies. Convallis malesuada aenean diam est quam.
    </p>
    <p>
      Erat quisque etiam consequat facilisis quisque; adipiscing mauris erat. Congue est dapibus sit eu mi non class felis ridiculus. Gravida nullam egestas suscipit parturient maximus malesuada. Sit viverra nascetur elementum tristique platea nullam molestie. Dui aptent fermentum condimentum efficitur eu ad phasellus nascetur. Litora himenaeos convallis magnis neque curae condimentum maximus. Fringilla dolor nulla dis ante congue nec montes ut.
    </p>
  </div>
</div>
`;
